packages/pulumi-aws/src/apps/api/ApiGateway.ts

```diff
@@ -1,50 +1,53 @@
 import * as aws from "@pulumi/aws";
 import * as pulumi from "@pulumi/pulumi";
-import { PulumiApp } from "@webiny/pulumi-sdk";
+import { defineAppModule, PulumiApp, PulumiAppModule } from "@webiny/pulumi-sdk";
 
 export interface ApiRouteParams {
     path: pulumi.Input<string>;
     method: pulumi.Input<string>;
     function: pulumi.Input<string>;
 }
 
-export type ApiGateway = ReturnType<typeof createApiGateway>;
+export type ApiGateway = PulumiAppModule<typeof ApiGateway>;
 
-export function createApiGateway(app: PulumiApp, routes: Record<string, ApiRouteParams>) {
-    const api = app.addResource(aws.apigatewayv2.Api, {
-        name: "api-gateway",
-        config: {
-            protocolType: "HTTP",
-            description: "Main API gateway"
-        }
-    });
+export const ApiGateway = defineAppModule({
+    name: "ApiGateway",
+    config(app: PulumiApp, routesConfig: Record<string, ApiRouteParams>) {
+        const api = app.addResource(aws.apigatewayv2.Api, {
+            name: "api-gateway",
+            config: {
+                protocolType: "HTTP",
+                description: "Main API gateway"
+            }
+        });
 
-    const defaultStage = app.addResource(aws.apigatewayv2.Stage, {
-        name: "default",
-        config: {
-            apiId: api.output.id,
-            autoDeploy: true
-        }
-    });
+        const stage = app.addResource(aws.apigatewayv2.Stage, {
+            name: "default",
+            config: {
+                apiId: api.output.id,
+                autoDeploy: true
+            }
+        });
 
-    const routesResult: Record<string, ReturnType<typeof createRoute>> = {};
+        const routes: Record<string, ReturnType<typeof createRoute>> = {};
 
-    for (const name of Object.keys(routes)) {
-        addRoute(name, routes[name]);
-    }
+        for (const name of Object.keys(routesConfig)) {
+            addRoute(name, routesConfig[name]);
+        }
 
-    return {
-        api,
-        defaultStage,
-        routes: routesResult,
-        addRoute
-    };
+        return {
+            api,
+            stage,
+            routes,
+            addRoute
```

Comment:
This is super cool 👍  Exposing customization methods like that... really digging that!

Reply:
And I like to define all of that kind of functions at the end of the function call, even after the `return` statement.
It's a cool way of utilizing hoisting to our advantage.